packages/react-dom/src/events/ReactDOMEventListener.js

```diff
@@ -148,16 +149,19 @@ function dispatchUserBlockingUpdate(
   container,
   nativeEvent,
 ) {
-  runWithPriority(
-    UserBlockingPriority,
-    dispatchEvent.bind(
-      null,
-      topLevelType,
-      eventSystemFlags,
-      container,
-      nativeEvent,
-    ),
-  );
+  // TODO: Double wrapping is necessary while we decouple Scheduler priority.
+  unstable_runWithPriority(UserBlockingPriority, () => {
```

Comment:
Since this is part of the renderer, we should be able to mutate the value directly without a wrapper function.

Not gonna block the PR over this, but does need to be fixed before we can land the experiment..

Reply:
Wait, also, isn't UserBlockingPriority the wrong type? Should be a lane priority.

Reply:
> isn't UserBlockingPriority the wrong type? Should be a lane priority.

It must have been buried, but I mentioned this in the comment [here](https://github.com/facebook/react/pull/19121#discussion_r449238705) - ReactDOM.runWithPriority accepts a `ReactPriorityLevel` instead of a `LanePriority` so that we don't have to leak the lane implementation while we do the decoupling. Internally in React, everything operates on the Lane priority.

Were you expecting that we would export all of the LanePriorities, and the `schedulerPriorityToLanePriority` conversion function from ReactFiberReconciler so we can pass LanePriorities to ReactDOM.runWithPriority?

> Since this is part of the renderer, we should be able to mutate the value directly without a wrapper function.

Do you mean that we wouldn't need the try/catch? That would require that we export the current update lane getter and setter from ReactFiberReconciler as well as all the LanePriorities like mentioned above. Right now we only export `runWithPriority (ReactPriorityLevel)` and `getCurrentUpdatePriority (ReactPriorityLevel)`.

Reply:
I think it's fine to leak the lane priorities to the renderer because it's part of the same build artifact. It doesn't cross any package boundary.

Leaking to via an API `ReactDOM.runWithPriority` is different because that's called by external code.

> ReactDOM.runWithPriority accepts a ReactPriorityLevel instead of a LanePriority so that we don't have to leak the lane implementation while we do the decoupling. Inside the reconciler, everything operates on the Lane priority.

I missed that comment originally but I don't buy that rationale. `ReactDOM.runWithPriority` is an unstable API that only exists internally if/when we need it to temporarily migrate external callers. We're going to delete it ASAP, so "leaking" update lane priorities is not a huge issue. But more importantly, the caller in this PR isn't external, so that's not relevant in this case.